server/src/services/order.service.ts

```diff
@@ -87,6 +97,12 @@ class OrderService {
 
     await OrderItem.bulkCreate(listOfOrderItems, { transaction });
 
+    for (const item of items) {
+      for (const category of item.item_categories) {
+        await scoreService.increment(transaction, user_id, category.category_id, ORDER_SCORE);
+      }
+    }
```

Comment:
오 이거 Promise.all 로 실행하면 더 빠르게 할 수 있지 않을까도 싶습니다.

```suggestion
    await Promise.all(
      items
        .flatMap((item) => item.item_categories)
        .map((category) => scoreService.increment(transaction, user_id, category.category_id, ORDER_SCORE))
    );
```

tsconfig에서 target을 es2019 이후(esnext 라던지)로 바꾼다면 [flatmap](https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap)이 사용 가능하고 간결하게 표현할 수 있을 듯 합니다!

Reply:
아니면 bulkCreate를 이용해 bulkUpdate 같은 효과를 만드는 것도... ㅎㅎ
https://stackoverflow.com/questions/54898994/bulkupdate-in-sequelize-orm

Reply:
깔끔하고 좋군요 ㅎㅎ 새벽에 급히 짜다보니 이중포문 돌려버린..!ㅎㅎ
추후에 리팩토링때 적용해보도록 하겠습니다! :)

Reply:
bulkCreate시 기존값을 알 수 없어서 업데이트할 수 없는 이슈가 있었습니다! Promise.all로 개선하겠습니다 :)